saleor/graphql/attribute/utils.py

```diff
@@ -234,10 +236,18 @@ def clean_input(
         return cleaned_input
 
     @staticmethod
-    def _clean_file_url(file_url: Optional[str]):
+    def _clean_file_url(file_url: Optional[str], error_class):
         # extract storage path from file URL
+        storage_root_url = build_absolute_uri(default_storage.base_url)
+        if file_url is not None and not file_url.startswith(
+            storage_root_url  # type: ignore
+        ):
+            raise ValidationError(
+                "The file_url must be the path to the default storage.",
+                code=error_class.INVALID.value,
+            )
         return (
-            re.sub(f"^{settings.MEDIA_URL}", "", urlparse(file_url).path)
+            re.sub(f"^{default_storage.base_url}", "", urlparse(file_url).path)
```

Comment:
two issues here:
 * `base_url` doesn't exist in all storage classes
 * in `FileSystemStorage` it tend to return `settings.MEDIA_URL`, which is just a path prefix (so will not catch protocol and host part in absolute URL)

Reply:
Yeah, but here I replacing the whole part of the protocol and host.